datalad/cmdline/support.py

```diff
@@ -0,0 +1,105 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+"""
+
+__docformat__ = "restructuredtext"
+
+from datalad.utils import swallow_outputs
+from datalad.api import wtf
+
+import os
+import sys
+
+from logging import getLogger
+
+lgr = getLogger("datalad.cmdline")
+
+# Disable helpme additional verbosity
+os.environ["MESSAGELEVEL"] = "QUIET"
+os.putenv("MESSAGELEVEL", "QUIET")
+
+# Set defaults for repository and title for issue
+default_repo = "datalad/datalad-helpme"
+default_title = "Test issue opened manually by helpme"
+
+
+def submit_helpme(title=None, traceback="", detail="", identifier=None, repo=None):
+    """Submit a request to the datalad-helpme repository at
+       https://github.com/datalad/datalad-helpme. If helpme isn't installed,
+       we skip this step. The basic submission includes the entire grab from
+       wtf (capturing system and library information) and an optional message.
+
+       Arguments:
+        - title (str)     : the title for the GitHub issue
+        - detail (str)    : any extra string content to include with the message.
+        - traceback (str) : the full traceback
+        - identifier (str): the identifier string (will use traceback if not defined)
+        - repo (str)      : GitHub repo (<username>/<repo>) to submit to.
+    """
+    title = title or default_title
+    repo = repo or default_repo
+
+    # If no identifier defined, use traceback
+    if identifier is None:
+        identifier = traceback
```

Comment:
Is there a special meaning of an empty string identifier?

Reply:
We check `if identifier is not None` so it can't be an empty string.